rescript/replace_taxonomy.py

```diff
@@ -0,0 +1,50 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2021, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from qiime2.plugin import MetadataColumn, Categorical
+import pandas as pd
+import re
+
+
+def make_substitutions_dict(taxonomy_replacement_map):
+    replacement_dict = taxonomy_replacement_map.to_series().to_dict()
+    return replacement_dict
+
+
+def make_regex(substitutions):
+    strings = substitutions.keys()
```

Comment:
if you only need the keys, then there is no need to transform to a dict above, just do:
```
strings = taxonomy_replacement_map.index
```
but there might be a more efficient way to do this (see below)